tests/PKSim.Tests/Presentation/WorkspaceLayoutCreatorSpecs.cs

```diff
@@ -88,4 +90,41 @@ public void should_mark_the_existing_layout_item_as_not_open()
          _workspace.WorkspaceLayout.LayoutItems.First(x => _existingWorkspaceLayout.LayoutItems.Contains(x)).WasOpenOnSave.ShouldBeFalse();
       }
    }
+
+   public class When_updating_workspace_layout_with_nonpersistent_subject_presenters : concern_for_WorkspaceLayoutCreator
+   {
+      private ISingleStartPresenter _presenter1;
+      private ISingleStartPresenter _presenter2;
+      private Individual _individual;
+      private ParameterIdentificationFeedback _feedback;
+      private IWorkspaceLayout _existingWorkspaceLayout;
+
+      protected override void Context()
+      {
+         base.Context();
+         _individual = new Individual().WithId("individual");
+         _feedback = new ParameterIdentificationFeedback(new ParameterIdentification());
+
+         _presenter1 = A.Fake<ISingleStartPresenter>();
+         A.CallTo(() => _presenter1.Subject).Returns(_individual);
+         _presenter2 = A.Fake<ISingleStartPresenter>();
+         A.CallTo(() => _presenter2.Subject).Returns(_feedback);
+         A.CallTo(() => _applicationController.OpenedPresenters()).Returns(new[] { _presenter1, _presenter2 });
+
+         _existingWorkspaceLayout = A.Fake<IWorkspaceLayout>();
+         _workspace.WorkspaceLayout = _existingWorkspaceLayout;
+         A.CallTo(() => _existingWorkspaceLayout.LayoutItems).Returns(new List<WorkspaceLayoutItem> { new WorkspaceLayoutItem() });
+      }
+
+      protected override void Because()
+      {
+         sut.SaveCurrentLayout();
+      }
+
+      [Observation]
+      public void should_return_an_object_containing_one_layout_item_for_each_opened_presenter_and_one_for_the_existing_layout_item()
```

Comment:
Test is ok but this caption is misleading. It does not create a layout item for each opened presenter! You are testing the opposite